lib/mutant/mutator/node/define.rb

```diff
@@ -13,11 +13,39 @@ class Define < self
         # @api private
         def dispatch
           emit_arguments_mutations
+          emit_optarg_mutations
           emit_body(N_RAISE)
           emit_body(nil)
           emit_body_mutations if body
         end
 
+        # Emit mutation with optional arguments as assignments in method
+        #
+        # @return [undefined]
+        #
+        # @api private
+        def emit_optarg_mutations
+          optargs, args = arguments.children.partition do |child|
+            child.type.equal?(:optarg)
+          end
+
+          return if optargs.empty?
+
+          lvasgns = optargs.map do |optarg|
+            s(:lvasgn, *optarg.children)
+          end
+
+          body_children = [*lvasgns, body].compact
+
+          new_body = if body_children.one?
+            body_children.first
+          else
+            s(:begin, *body_children)
+          end
```

Comment:
I think that's a fine approach.

OT, but I wonder if there's room for a library where you could hand it a `Parser::AST::Node` and it would do some basic minimization and produce the simplest possible equivalent AST. I wouldn't necessarily want `unparser` to do that work, since it would be unexpected if it changed the structure.

If we had this, then we could build the AST using the most straight-forward way, and then minimize it before emitting it.

@mbj wdyt?